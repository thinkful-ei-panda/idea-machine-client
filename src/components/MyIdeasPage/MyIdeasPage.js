import React from 'react'
import Results from '../IdeaResults/IdeaResults'
import config from '../../config'

class MyIdeasPage extends React.Component {
  state = {
    results: [{id:1,title:'some title',content:'some content'}],
    error:null,
  }

  componentDidMount(){
    fetch(`${config.API_ENDPOINT}/ideas/my-ideas`)
    .then(res => (!res.ok)
    ?res.json().then(e => Promise.reject(e))
    :res.json())
    .then(res => {
      this.setState({results:res})
    })
    .catch(error => {
      this.setState({error:error})
    })
  }

  render() {
    const {results} = this.state
    return (
      <div>
        My Ideas Page
        {results.length !== 0 && <Results results={results} />}
      </div>
    )
  }  
}

export default MyIdeasPage