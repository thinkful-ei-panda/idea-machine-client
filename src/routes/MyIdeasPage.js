import React from 'react';
import MyIdeas from '../components/MyIdeas/MyIdeas';

class MyIdeasPage extends React.Component {

  render() {
    return (
      <div className='MyIdeasPage'>
        <h2 hidden>My Ideas</h2>
        <MyIdeas {...this.props}/>
      </div>
    );
  }
}

export default MyIdeasPage
