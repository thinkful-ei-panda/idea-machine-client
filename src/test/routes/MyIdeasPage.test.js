import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import MyIdeasPage from '../../routes/MyIdeasPage';
describe(`MyIdeasPage Component`, () => {
    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>
                    <MyIdeasPage />
                </BrowserRouter>
                , div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <MyIdeasPage />
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})