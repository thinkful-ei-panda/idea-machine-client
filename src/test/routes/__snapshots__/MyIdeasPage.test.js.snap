// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyIdeasPage Component Snapshot test Renders the UI as expected 1`] = `
<div
  className="MyIdeasPage"
>
  <h2
    hidden={true}
  >
    My Ideas
  </h2>
  <div
    className="buttonContainer-add-idea"
  >
    <a
      href="/add-idea"
      onClick={[Function]}
    >
      <button>
        Add Idea
      </button>
    </a>
  </div>
</div>
`;
